### ISSUE TITLE
health_check.php fails if any database cache engine configured

### ISSUE THREAD

**Issue by ilnytskyi** [magento/magento2#25669]
### Preconditions (*)

1. Configure any cache to database in env.php
2.3-develop
```
    'cache' => [
        'frontend' => [
            '{some_frontend_cache}' => [
                'backend' => 'database',
                'backend_options' => [],
                'id_prefix' => 'U_'
            ]
        ],
        'type' => [
            '{any_cache_type_here}' => [
                'frontend' => '{some_frontend_cache}'
            ]
        ]
    ]
```
2.

### Steps to reproduce (*)

1. configure any cache type to database
2.

### Expected result (*)

1. health_check.php works
2. 

### Actual result (*)

1. health_check.php fails since can't find `database` class in 
```
$backend = new $cacheBackendClass($cacheConfig[ConfigOptionsListConstants::CONFIG_PATH_BACKEND_OPTIONS]);
```
2. health_check.php fails since can't create instance of `\Magento\Framework\Cache\Backend\Database` and throws expeditions from here 
`\Magento\Framework\Cache\Backend\Database::__construct`
```
'Option "adapter" should be declared and extend \Magento\Framework\DB\Adapter\AdapterInterface!'
Option "data_table" or "data_table_callback" should be declared!
Option "tags_table" or "tags_table_callback" should be declared!
'DB Adapter should be declared and extend \Magento\Framework\DB\Adapter\AdapterInterface'
```

however the cache works just uses default connection
![Selection_210](https://user-images.githubusercontent.com/3192770/69239684-1649e780-0b9b-11ea-8841-c4a5863b152f.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Echo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29016` were created

**Issue Available**: @engcom-Echo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by andrewbess** [comment]
#cdkharkiv19nov

**Comment by magento-engcom-team** [comment]
Hi @ilnytskyi. Thank you for your report. 
The issue has been fixed in magento/magento2#25722 by @andrewbess in 2.4-develop branch
Related commit(s):
 - [5c121e20f909d666079654c6aa7ff325cc094278](https://github.com/magento/magento2/commit/5c121e20f909d666079654c6aa7ff325cc094278)
 - [3905038c09ee3c5f7a9dac63f5f28070c1197434](https://github.com/magento/magento2/commit/3905038c09ee3c5f7a9dac63f5f28070c1197434)
 - [e59ee3a1aafc648d7c297660e986bafbaee0b54f](https://github.com/magento/magento2/commit/e59ee3a1aafc648d7c297660e986bafbaee0b54f)
 - [6633cc22aa266a0399a60d6a7b94f8d8841004be](https://github.com/magento/magento2/commit/6633cc22aa266a0399a60d6a7b94f8d8841004be)

The fix will be available with the upcoming 2.4.0 release.
